packages/db2/src/connection-manager.ts

```diff
@@ -22,16 +24,15 @@ export interface Db2Connection extends Connection, Db2.Database {}
  * @private
  */
 export class Db2ConnectionManager extends AbstractConnectionManager<Db2Dialect, Db2Connection> {
-  readonly #lib: typeof Db2;
+  readonly #lib: IbmDbModule;
```

Comment:
Should all the dialects use `private` instead of a mix of `#` and `private`.

```suggestion
  private readonly lib: IbmDbModule;
```

Reply:
In my opinion, when we can enforce hard private, we should go for that one. When we can't due to a technical issue, we should use `private`, and prefix the field with `_`. When even `private` is not possible, just prefixing with `_` as a last resort

Reply:
Good plan.
By adding the `[dialectName]Module` to the options, you can remove the need for the `lib` to be `private readonly lib` as this can be mocked in the tests.

If the below changes are made, it would mean all dialects can have the `lib` set to hard private.

https://github.com/sequelize/sequelize/blob/73bb04e64ac2340436cc271d863f2af37de5897b/packages/core/test/unit/dialects/mssql/connection-manager.test.ts#L1-L49

```ts
import type { Options } from '@sequelize/core';
import { ConnectionError, Sequelize } from '@sequelize/core';
import type { MsSqlDialect } from '@sequelize/mssql';
import type { RequiredBy } from '@sequelize/utils';
import { assert, expect } from 'chai';
import sinon from 'sinon';
import * as Tedious from 'tedious';
import { getTestDialect } from '../../../support';

const dialect = getTestDialect();

type TestConnection = Omit<Tedious.Connection, 'once' | 'removeListener' | 'on'> & {
  once(event: string, cb: () => void): void;
  removeListener(): void;
  on(): void;
};

describe('[MSSQL Specific] Connection Manager', () => {
  if (dialect !== 'mssql') {
    return;
  }

  let config: RequiredBy<Options<MsSqlDialect>, 'dialectOptions'>;
  let instance: Sequelize<MsSqlDialect>;
  let Connection: Partial<TestConnection>;
  let connectionStub: sinon.SinonStub;

  beforeEach(() => {
    config = {
      dialect: 'mssql',
      database: 'none',
      username: 'none',
      password: 'none',
      host: 'localhost',
      port: 2433,
      pool: {},
      dialectOptions: {
        domain: 'TEST.COM',
      },
      tediousModule: Tedious,
    };
    Connection = {};
    connectionStub = sinon.stub(config, 'tediousModule').value({
      Connection: function fakeConnection() {
        return Connection;
      },
    });
    instance = new Sequelize(config);
  });
```

Reply:
Done! (slightly differently)